src/modules/oh-indicators-list/network-layer/sseClient.ts

```diff
@@ -1,18 +1,41 @@
+interface IEventHandlers {
+  onMessageHandler: (event: MessageEvent) => void;
+  onOpenHandler: () => void;
+  onErrorHandler: () => void;
+}
 export interface IEventSourceInitializer {
   init: (
-    eventName: string,
-    eventHandlers: {
-      onMessageHandler: (event: MessageEvent) => void;
-      onOpenHandler: (eventName: string) => void;
-      onErrorHandler: (e: any) => void;
-    },
+    url: string,
+    eventHandlers: IEventHandlers,
   ) => void;
 }
 
 export interface IEventSourceFinalizer {
-  destroy: (eventName: string) => void;
+  destroy: () => void;
 }
 
-export default class {
-    // TODO implementation
-}
\ No newline at end of file
+export const parseStreamingResponse = (
+  incomingMessage: MessageEvent,
+): { name: string; value: string } => {
+  const data = JSON.parse(incomingMessage.data);
+  const [, , name] = data.topic.split('/');
+  const { value } = JSON.parse(data.payload);
+  return { name, value };
+};
+
+export default class SseClient {
+  source: EventSource | undefined;
+  init(url: string, eventHandlers: IEventHandlers) {
```

Comment:
What if I want to open multiple connections during the app's lifetime? 🙃

Reply:
We can have only 1 connection, we can close it and create again. No multiple connections in a row.

Reply:
The middleware has to provide scalability, it may come in handy in the future - the possibility to independently subscribe to different events (even if it won't be needed). Let's add

Reply:
Just keep a dictionary of all the subscriptions and the unsub/destroy would require event/url name